### ISSUE TITLE
e2e_fluentbit_output_test.go failed in the CI

### ISSUE THREAD

**Issue by shashankram** [openservicemesh/osm#3138]
**Bug description**:
`e2e_fluentbit_output_test.go` failed in the CI with the following error:
```
• Failure [48.295 seconds]
[Tier 2][Bucket 2] Test deployment of Fluent Bit sidecar
/home/runner/work/osm/osm/tests/framework/common.go:132
  Fluent Bit output
  /home/runner/work/osm/osm/tests/e2e/e2e_fluentbit_output_test.go:25
    Forwards correctly filtered logs to stdout [It]
    /home/runner/work/osm/osm/tests/e2e/e2e_fluentbit_output_test.go:26

    Expected
        <string>: Fluent Bit v1.6.4
        * Copyright (C) 2019-2020 The Fluent Bit Authors
        * Copyright (C) 2015-2018 Treasure Data
        * Fluent Bit is a CNCF sub-project under the umbrella of Fluentd
        * https://fluentbit.io
        
        [2021/04/12 21:36:30] [ info] [engine] started (pid=1)
        [2021/04/12 21:36:30] [ info] [storage] version=1.0.6, initializing...
        [2021/04/12 21:36:30] [ info] [storage] in-memory
        [2021/04/12 21:36:30] [ info] [storage] normal synchronization mode, checksum disabled, max_chunks_up=128
        [2021/04/12 21:36:30] [ info] [sp] stream processor started
        [2021/04/12 21:36:30] [ info] [input:tail:tail.0] inotify_fs_add(): inode=842690 watch_fd=1 name=/var/log/containers/osm-controller-76988c545b-5vm85_osm-system_osm-controller-77461716af9f3006e1290d34a870c0e2bed7fa08bc7f71239e76651524c6c014.log
        
    to contain substring
        <string>: "level":"info"

    /home/runner/work/osm/osm/tests/e2e/e2e_fluentbit_output_test.go:49
```

Logs:
[logs_e2e_fluentbit_output_test.zip](https://github.com/openservicemesh/osm/files/6299977/logs_e2e_fluentbit_output_test.zip)

**Affected area (please mark with X where applicable)**:
- Install                    [ ]
- SMI Traffic Access Policy  [ ]
- SMI Traffic Specs Policy   [ ]
- SMI Traffic Split Policy   [ ]
- Permissive Traffic Policy  [ ]
- Ingress                    [ ]
- Egress                     [ ]
- Envoy Control Plane        [ ]
- CLI Tool                   [ ]
- Metrics                    [ ]
- Certificate Management     [ ]
- Sidecar Injection          [ ]
- Logging                    [ ]
- Debugging                  [ ]
- Tests                      [X]
- Demo                       [ ]
- CI System                  [ ]

**Comment by shashankram** [comment]
Another CI run for this test failed recently:
```
   Expected
        <string>: Fluent Bit v1.6.4
        * Copyright (C) 2019-2020 The Fluent Bit Authors
        * Copyright (C) 2015-2018 Treasure Data
        * Fluent Bit is a CNCF sub-project under the umbrella of Fluentd
        * https://fluentbit.io
        
        [2021/05/07 00:03:59] [ info] [engine] started (pid=1)
        [2021/05/07 00:03:59] [ info] [storage] version=1.0.6, initializing...
        [2021/05/07 00:03:59] [ info] [storage] in-memory
        [2021/05/07 00:03:59] [ info] [storage] normal synchronization mode, checksum disabled, max_chunks_up=128
Error:  [2021/05/07 00:03:59] [error] [plugins/in_tail/tail_file.c:617 errno=13] Permission denied
Error:  [2021/05/07 00:03:59] [error] [input:tail:tail.0] cannot open /var/log/containers/osm-controller-84b7cb747f-l2v8m_osm-system_osm-controller-6e45557ef3437381c0f7765c59f0b612b85c8080964acd411181a307073ef8ea.log
        [2021/05/07 00:03:59] [ info] [sp] stream processor started
        
    to contain substring
        <string>: "level":"info"

    /home/runner/work/osm/osm/tests/e2e/e2e_fluentbit_output_test.go:49
```

Logs:
[e2e_fluentbit_failure.zip](https://github.com/openservicemesh/osm/files/6438206/e2e_fluentbit_failure.zip)

**Comment by SanyaKochhar** [comment]
@shashankram thanks for sharing logs. This permission denied error looks new, I'll take a look

**Comment by shashankram** [comment]
Seeing consistent failures in the CI now:

Last 2 runs failed with the below issue:
https://github.com/openservicemesh/osm/actions/runs/818598279
https://github.com/openservicemesh/osm/actions/runs/821475767

```
STEP: [Cleanup] waiting for osmTest:1620417342 test NS cleanup

• Failure [36.279 seconds]
[Tier 2][Bucket 2] Test deployment of Fluent Bit sidecar
/home/runner/work/osm/osm/tests/framework/common.go:136
  Fluent Bit output
  /home/runner/work/osm/osm/tests/e2e/e2e_fluentbit_output_test.go:25
    Forwards correctly filtered logs to stdout [It]
    /home/runner/work/osm/osm/tests/e2e/e2e_fluentbit_output_test.go:26

    Expected
        <string>: Fluent Bit v1.6.4
        * Copyright (C) 2019-2020 The Fluent Bit Authors
        * Copyright (C) 2015-2018 Treasure Data
        * Fluent Bit is a CNCF sub-project under the umbrella of Fluentd
        * https://fluentbit.io
        
        [2021/05/07 20:04:21] [ info] [engine] started (pid=1)
        [2021/05/07 20:04:21] [ info] [storage] version=1.0.6, initializing...
        [2021/05/07 20:04:21] [ info] [storage] in-memory
        [2021/05/07 20:04:21] [ info] [storage] normal synchronization mode, checksum disabled, max_chunks_up=128
Error:  [2021/05/07 20:04:21] [error] [plugins/in_tail/tail_file.c:617 errno=13] Permission denied
Error:  [2021/05/07 20:04:21] [error] [input:tail:tail.0] cannot open /var/log/containers/osm-controller-84b7cb747f-zjm9s_osm-system_osm-controller-d0ab35e719d0a3b479c533498c5b570f452b5f01c4ce152c11826a6ffb4a03b3.log
        [2021/05/07 20:04:21] [ info] [sp] stream processor started
        
    to contain substring
        <string>: "level":"info"

    /home/runner/work/osm/osm/tests/e2e/e2e_fluentbit_output_test.go:49
```

**Comment by shashankram** [comment]
The more recent failures are due to the addition of security contexts to the OSM pods introduced in https://github.com/openservicemesh/osm/pull/3299, which is different than the original issue this bug was filed for.

@ksubrmnn would you mind checking this, security contexts are being added even when PSP is disabled, which seems incorrect.

**Comment by shashankram** [comment]
Another occurrence of this failure https://github.com/openservicemesh/osm/runs/3205751883?check_suite_focus=true:

```
• Failure [86.685 seconds]
[Tier 2][Bucket 3] Test deployment of Fluent Bit sidecar
/home/runner/work/osm/osm/tests/framework/common.go:89
  Fluent Bit output
  /home/runner/work/osm/osm/tests/e2e/e2e_fluentbit_output_test.go:25
    Forwards correctly filtered logs to stdout [It]
    /home/runner/work/osm/osm/tests/e2e/e2e_fluentbit_output_test.go:26

    Expected
        <string>: Fluent Bit v1.6.4
        * Copyright (C) 2019-2020 The Fluent Bit Authors
        * Copyright (C) 2015-2018 Treasure Data
        * Fluent Bit is a CNCF sub-project under the umbrella of Fluentd
        * https://fluentbit.io
        
        [2021/07/30 21:34:21] [ info] [engine] started (pid=1)
        [2021/07/30 21:34:21] [ info] [storage] version=1.0.6, initializing...
        [2021/07/30 21:34:21] [ info] [storage] in-memory
        [2021/07/30 21:34:21] [ info] [storage] normal synchronization mode, checksum disabled, max_chunks_up=128
        [2021/07/30 21:34:21] [ info] [sp] stream processor started
        [2021/07/30 21:34:21] [ info] [input:tail:tail.0] inotify_fs_add(): inode=544199 watch_fd=1 name=/var/log/containers/osm-controller-79df4fd889-knht4_osm-system_osm-controller-8ec47d95d775da99e3cdba3767a185764d8bc43a20e6cc900279f05c6204788d.log
        
    to contain substring
        <string>: "level":"info"

    /home/runner/work/osm/osm/tests/e2e/e2e_fluentbit_output_test.go:52
```

**Comment by snehachhabria** [comment]
The test is falky and I am seeing failures too, @SanyaKochhar please could you look into this and address the issue if possible

**Comment by shashankram** [comment]
Another occurrence of this failure:

![image](https://user-images.githubusercontent.com/21697719/132718333-3c23a0b8-6feb-433a-9e04-e0460a9b8d1d.png)

**Comment by michelleN** [comment]
+1 I've also seen this issue.

**Comment by snehachhabria** [comment]
I just saw the issue too, I think we definitely need to fix it for v1.0.0 or disable the test if we aren't planning on maintaining it going forward

**Comment by SanyaKochhar** [comment]
Closing this with #4124 by disabling the test. Can't seem to find the root cause -- looking at some of the logs, the "level: info" was emitted by the osm controller and seen in the fluent bit container logs. Could be flakey due to a timing issue with wait time here https://github.com/openservicemesh/osm/blob/24c4852ace03e43548713a638025cc0e6fed9c71/tests/e2e/e2e_fluentbit_output_test.go#L63 but doesn't seem likely
